assignment-section-05/generate_test_data.py

```diff
@@ -0,0 +1,26 @@
+from pathlib import Path
+
+from classification_model.config.core import config, DATASET_DIR
+from classification_model.processing.data_manager import _load_raw_dataset
+
+
+def generate_data_data() -> None:
+    """
+    Generate train.csv and test.csv files.
+
+    The titanic dataset does not come split into train and test files, so we
+    do this ourselves. Note, this requires you to have downloaded the original
+    datafile from here: https://www.openml.org/data/get_csv/16826755/phpMYEkMl
+    and renamed it to match your config raw_data_file name.
+    """
+    data = _load_raw_dataset(file_name=config.app_config.raw_data_file)
+    df_len = len(data)
+    row_split_index = round(df_len*(1-config.model_config.test_size))
```

Comment:
the split done by `train_test_split` is at random. Here you are doing top/bottom as train test, so it is not the same.

Not sure what is the intention of this function and I don't think students will intuitively generate something like this.

If this is needed, it needs to be made clear before they start the assignment. Probably provide a pre-drafted template.

Reply:
deleted file